src/examples/mnist/MNIST_SP.cpp

```diff
@@ -0,0 +1,240 @@
+        break;
+    }
+  }
+
+  SDR input({28, 28, 1});
+  SpatialPoolerExtended sp(
+    /* inputDimensions */           input.dimensions,
+    /* columnDimensions */          {10, 10, 120},
+    /* potentialRadius */           4,
+    /* potentialPct */              .9,
+    /* wrapAround */                true,
+    /* localAreaDensity */          .015,
+    /* stimulusThreshold */         28,
+    /* synPermInactiveDec */        .00928,
+    /* synPermActiveInc */          .032,
+    /* synPermConnected */          .422,
+    /* dutyCyclePeriod */           1402,
+    /* minPctOverlapDutyCycles */   0.,
+    /* seed */                      Random()(),
+    /* spVerbosity */               verbosity);
+
+  SDR columns({sp.getNumColumns()});
+  SDR_Metrics columnStats(columns, 1402);
+
+  SDRClassifier clsr(
+    /* steps */         {0},
+    /* alpha */         .001,
+    /* actValueAlpha */ .3,
+                        verbosity);
+
+  // Train
+  auto train_images = read_mnist_images("./mnist_data/train-images-idx3-ubyte");
+  auto train_labels = read_mnist_labels("./mnist_data/train-labels-idx1-ubyte");
+  if(verbosity)
+    cout << "Training for " << (train_dataset_iterations * train_labels.size())
+         << " cycles ..." << endl;
+  for(auto i = 0u; i < train_dataset_iterations; i++) {
+    // Shuffle the training data.
+    vector<UInt> index( train_labels.size() );
+    for(auto s = 0u; s < train_labels.size(); s++)
+        index[s] = s;
+    Random().shuffle( index.begin(), index.end() );
+
+    for(auto s = 0u; s < train_labels.size(); s++) {
+      // Get the input & label
+      UInt *image = train_images[ index[s] ];
+      UInt label  = train_labels[ index[s] ];
+
+      // Compute & Train
+      input.setDense( image );
+      sp.compute(input, true, columns);
+      ClassifierResult result;
+      clsr.compute(sp.getIterationNum(), columns.getFlatSparse(),
+        /* bucketIdxList */   {label},
+        /* actValueList */    {(Real)label},
+        /* category */        true,
+        /* learn */           true,
+        /* infer */           false,
+                              &result);
+      if( verbosity and sp.getIterationNum() % 1000 == 0 )
+        cout << "." << flush;
+    }
+  }
+  if( verbosity ) cout << endl;
+
+  cout << columnStats;
+
+  // Test
+  auto test_images  = read_mnist_images("./mnist_data/t10k-images-idx3-ubyte");
+  auto test_labels  = read_mnist_labels("./mnist_data/t10k-labels-idx1-ubyte");
+  Real score = 0;
+  UInt n_samples = 0;
+  if(verbosity)
+    cout << "Testing for " << test_labels.size() << " cycles ..." << endl;
+  for(UInt i = 0; i < test_labels.size(); i++) {
+    // Get the input & label
+    UInt *image = test_images[i];
+    UInt label  = test_labels[i];
+
+    // Compute
+    input.setDense( image );
+    sp.compute(input, false, columns);
+    ClassifierResult result;
+    clsr.compute(sp.getIterationNum(), columns.getFlatSparse(),
```

Comment:
please add timer for SP, Classifier, so we can measure performance. Same as in #138